src/triage/risklist/__init__.py

```diff
@@ -0,0 +1,236 @@
+    from triage_metadata.experiments
+    join triage_metadata.experiment_models using (experiment_hash)
+    join triage_metadata.models using (model_hash)
+    where model_id = %s
+    '''
+    (config,) = db_engine.execute(get_experiment_query, model_id).first()
+    return config
+    
+
+def train_matrix_info_from_model_id(db_engine, model_id):
+    """Get original train matrix information from model_id 
+    Args:
+            db_engine (sqlalchemy.db.engine)
+            model_id (int) The id of a given model in the database
+
+    Returns: (str, dict) matrix uuid and matrix metadata
+    """
+    get_train_matrix_query = """
+        select matrix_uuid, matrices.matrix_metadata
+        from triage_metadata.matrices
+        join triage_metadata.models on (models.train_matrix_uuid = matrices.matrix_uuid)
+        where model_id = %s
+    """
+    return db_engine.execute(get_train_matrix_query, model_id).first()
+
+
+def get_feature_names(aggregation, matrix_metadata):
+    """Returns a feature group name and a list of feature names from a SpacetimeAggregation object"""
+    feature_prefix = aggregation.prefix
+    logger.spam("Feature prefix = %s", feature_prefix)
+    feature_group = aggregation.get_table_name(imputed=True).split('.')[1].replace('"', '')
+    logger.spam("Feature group = %s", feature_group)
+    feature_names_in_group = [f for f in matrix_metadata['feature_names'] if re.match(f'\\A{feature_prefix}', f)]
+    logger.spam("Feature names in group = %s", feature_names_in_group)
+    
+    return feature_group, feature_names_in_group
+
+def get_feature_needs_imputation_in_train(aggregation, feature_names):
+    """Returns features that needs imputation from training data
+    Args:
+        aggregation (SpacetimeAggregation)
+        feature_names (list) A list of feature names
+    """
+    features_imputed_in_train = [
+        f for f in set(feature_names)
+        if not f.endswith('_imp') 
+        and aggregation.imputation_flag_base(f) + '_imp' in feature_names
+    ]
+    logger.spam("Features imputed in train = %s", features_imputed_in_train)
+    return features_imputed_in_train
+
+
+def get_feature_needs_imputation_in_production(aggregation, db_engine):
+    """Returns features that needs imputation from production
+    Args:
+        aggregation (SpacetimeAggregation)
+        db_engine (sqlalchemy.db.engine)
+    """
+    with db_engine.begin() as conn:
+        nulls_results = conn.execute(aggregation.find_nulls())
+    
+    null_counts = nulls_results.first().items()
+    features_imputed_in_production = [col for (col, val) in null_counts if val > 0]
+    
+    return features_imputed_in_production
+
+
+def generate_risk_list(db_engine, project_storage, model_id, as_of_date):
+    """Generate the risk list based model_id and as_of_date
+
+    Args:
+            db_engine (sqlalchemy.db.engine)
+            project_storage (catwalk.storage.ProjectStorage)
```

Comment:
Possible to let this take either a `ProjectStorage` object or a path? I feel like the later would be a little more natural when using the python interface directly.

Reply:
I could just make it take a path. ProjectStorage is right now mostly useful internally to Triage code. Making it take the path is consistent with Experiment.

Reply:
Thanks -- trying to keep the interface pretty consistent with `Experiment` feels like a good idea to me.